TowerForge/TowerForge/Firebase/Authentication/AuthenticationProvider.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  AuthenticationProvider.swift
+//  TowerForge
+//
+//  Created by Vanessa Mae on 08/04/24.
+//
+
+import Foundation
+
+class AuthenticationProvider {
+    private let authenticationManager: AuthenticationManager
+    private var observers = [AuthenticationDelegate]()
+
+    init(authenticationManager: AuthenticationManager = AuthenticationManager()) {
+        self.authenticationManager = authenticationManager
+        self.authenticationManager.setListener(delegate: self)
+    }
+    func login(email: String, password: String, completion: @escaping (AuthenticationData?, Error?) -> Void) {
+        self.authenticationManager.loginUser(email: email,
+                                             password: password,
+                                             completion: completion)
+        print("In provider: login")
```

Comment:
I think can change this to use Logger instead of print so we can turn off debug printing before production release.
```suggestion
        Logger.log("In provider: login", self)
```